x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -82,19 +88,39 @@ export class CloudPlugin implements Plugin<CloudSetup> {
   public start(coreStart: CoreStart, { security }: CloudStartDependencies) {
     const { deployment_url: deploymentUrl, base_url: baseUrl } = this.config;
     coreStart.chrome.setHelpSupportUrl(ELASTIC_SUPPORT_LINK);
-    if (baseUrl && deploymentUrl) {
-      coreStart.chrome.setCustomNavLink({
-        title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
-          defaultMessage: 'Manage this deployment',
-        }),
-        euiIconType: 'arrowLeft',
-        href: getFullCloudUrl(baseUrl, deploymentUrl),
-      });
-    }
 
-    if (security && this.isCloudEnabled) {
-      const userMenuLinks = createUserMenuLinks(this.config);
-      security.navControlService.addUserMenuLinks(userMenuLinks);
-    }
+    this.checkIfAuthorizedForLinks().then((authorized) => {
+      if (authorized && baseUrl && deploymentUrl) {
+        coreStart.chrome.setCustomNavLink({
+          title: i18n.translate('xpack.cloud.deploymentLinkLabel', {
+            defaultMessage: 'Manage this deployment',
+          }),
+          euiIconType: 'arrowLeft',
+          href: getFullCloudUrl(baseUrl, deploymentUrl),
+        });
+      }
+
+      if (authorized && security && this.isCloudEnabled) {
+        const userMenuLinks = createUserMenuLinks(this.config);
+        security.navControlService.addUserMenuLinks(userMenuLinks);
+      }
+    });
+  }
+
+  /**
+   * Determines if the current user should see links back to Cloud.
+   * This isn't a true authorization check, but rather a heuristic to
+   * see if the current user is *likely* a cloud deployment administrator.
+   *
+   * At this point, we do not have enough information to reliably make this determination,
+   * but we do know that all cloud deployment admins are superusers by default.
+   */
+  private async checkIfAuthorizedForLinks() {
+    // Security plugin is disabled
+    if (!this.authenticatedUserPromise) return true;
+    // Otherwise check roles. If user is not defined due to an unexpected error, then fail *open*.
+    // Cloud admin console will always perform the actual authorization checks.
+    const user = await this.authenticatedUserPromise;
+    return user?.roles.includes('superuser') ?? true;
```

Comment:
__question:__ sorry if I missed that in https://github.com/elastic/kibana/issues/97308, but for how long we expect this workaround to exist? If we're not sure, I'm wondering if we may want to leave an escape hatch and expose something like `xpack.cloud.roles: schema.arrayOf(schema.string(), { defaultValue: ['superuser'] })` config value, so that in the worst case users could create an empty "marker" role if they need to.... I don't have a strong opinion on that, so feel free to ignore.

Reply:
This workaround will likely exist for a while - it's really up to the cloud team's roadmap to prioritize an API or similar mechanism for us to leverage.

I think it's premature to add an escape hatch, because we don't know what it would look like just yet. It's possible that we would instead want something under cloud's control, such as inspecting the user's metadata that their SSO realm attaches

Reply:
> I think it's premature to add an escape hatch, because we don't know what it would look like just yet. It's possible that we would instead want something under cloud's control, such as inspecting the user's metadata that their SSO realm attaches

That makes sense, thanks. Works for me as long as it works for Cloud.